src/fse/fse_decoder.rs

```diff
@@ -125,13 +147,20 @@ impl From<GetBitsError> for FSEDecoderError {
     }
 }
 
+/// A single entry in an FSE table.
 #[derive(Copy, Clone)]
 pub struct Entry {
+    /// This value is used as an offset value, and it is added
+    /// to a value read from the stream to determine the next state value.
     pub base_line: u32,
+    /// How many bits should be read from the stream when decoding this entry.
     pub num_bits: u8,
+    /// The byte to replace this entry with during decoding (I think?).
```

Comment:
/// The byte that should be put in the decode output when encountering this state